backend/coreapp/views/project.py

```diff
@@ -134,6 +189,82 @@ def attempts(self, request, **kwargs):
         else:
             raise Exception("Unsupported method")
 
+    @action(detail=True, methods=["POST"])
+    def pr(self, request, pk):
+        scratch_slugs: list[str] = request.data.get("scratch_slugs", [])
+        if not scratch_slugs:
+            return  # TODO: raise exception
+
+        # TODO: make unique by GETting the branch
+        suffix = "".join(
+            random.choice(string.ascii_lowercase + string.digits) for _ in range(5)
+        )
+        fork_branch = f"decompme_GEN_{suffix}"
+
+        files_to_funcs: dict[str, list[str]] = {}
+        for scratch_slug in scratch_slugs:
+            try:
+                scratch = Scratch.objects.get(slug=scratch_slug)
+            except Scratch.DoesNotExist:
+                # TODO new exception type
+                raise Exception(f"Scratch slug={scratch_slug} does not exist")
+
+            if scratch.owner != request.profile:
+                raise ScratchNotOwnerException()
```

Comment:
I reckon we should allow this and instead put a `Co-authored by: <username> (<email>)` in the commit message body if you aren't the owner. I'm envisioning project maintainers batch-creating a PR of scratches where the owners of said scratches on decomp.me didn't create a PR for whatever reason